compare256_rle.h

```diff
@@ -88,7 +89,11 @@ static inline uint32_t compare256_rle_unaligned_32(const uint8_t *src0, const ui
 
         diff = sv ^ mv;
         if (diff) {
+#if BYTE_ORDER == LITTLE_ENDIAN
             uint32_t match_byte = __builtin_ctz(diff) / 8;
```

Comment:
Would it be worth using macros for this so that the endianness checks (and the builtin availability checks) can be performed in one place?

Reply:
As of right now this is only ever used in two files in 2 functions. Maybe it'd be worth a second PR, though.